components/builder-jobsrv/src/data_store.rs

```diff
@@ -416,27 +454,46 @@ impl DataStore {
             jobsrv::JobState::Rejected => "Rejected",
             jobsrv::JobState::Failed => "Failed",
         };
-        
+
         // Note: the following fields may all be NULL. As currently
         // coded, if they are NULL, then the corresponding fields in
         // the database will also be updated to be NULL. This should
         // be OK, though, because they shouldn't be changing anyway.
         let build_started_at = match job.has_build_started_at() {
             true => Some(DateTime::<UTC>::from_str(job.get_build_started_at()).unwrap()),
-            false => None
+            false => None,
         };
         let build_finished_at = match job.has_build_finished_at() {
             true => Some(DateTime::<UTC>::from_str(job.get_build_finished_at()).unwrap()),
-            false => None
+            false => None,
         };
         let ident = match job.has_package_ident() {
             true => Some(job.get_package_ident().to_string()),
-            false => None
+            false => None,
         };
 
-        conn.execute("SELECT update_job_v1($1, $2, $3, $4, $5)",
-                     &[&job_id, &job_state, &build_started_at, &build_finished_at, &ident])
-            .map_err(Error::JobSetState)?; 
+        if job.has_error() {
+            let err_code = job.get_error().get_code() as i32;
+            let err_msg = job.get_error().get_msg();
+
+            conn.execute("SELECT update_job_v2($1, $2, $3, $4, $5, $6, $7)",
+                         &[&job_id,
+                           &job_state,
+                           &build_started_at,
+                           &build_finished_at,
+                           &ident,
+                           &err_code,
+                           &err_msg])
+                .map_err(Error::JobSetState)?;
+        } else {
+            conn.execute("SELECT update_job_v1($1, $2, $3, $4, $5)",
```

Comment:
is `update_job_v1` sticking around for non error cases? If so, maybe we should rename the v2 function to something else instead of versioning it.